src/v/redpanda/monitor_unsafe_log_flag.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "redpanda/monitor_unsafe_log_flag.h"
+
+#include "utils/utf8.h"
+
+monitor_unsafe_log_flag::monitor_unsafe_log_flag(
+  cluster::cluster_version original_version)
+  : _original_version(original_version)
+  , _legacy_permit_unsafe_log_operation(
+      config::shard_local_cfg().legacy_permit_unsafe_log_operation.bind()) {
+    handle_flag_change();
+    _legacy_permit_unsafe_log_operation.watch(
```

Comment:
does this end up in having each local sharded service signing up to observe this event and all of them will get notified upon change of the flag?

Reply:
yes

Reply:
Why couldn't consumers of this information just check the flags value itself?

Reply:
(I assume "Flag Value" `X` means "whatever" in the table.)

I wonder what is wrong with this flag controlling the unsafe chars behaviour in newer clusters? Just trying to understand the leveraging for all the complications that are consequent from there.

Reply:
> (I assume "Flag Value" X means "whatever" in the table.)

Yes.

> I wonder what is wrong with this flag controlling the unsafe chars behaviour in newer clusters? Just trying to understand the leveraging for all the complications that are consequent from there.

Good question and I didn't provide any context, that's my fault.  At some point in the near future (in the next release cycle or two) this flag will go away and control characters will be permanently disallowed.  So the reason we're not 'honoring' the flag in new (v23.2.1 and above) clusters is because we don't want the user to be able to add items into Redpanda that do contain control characters.

Reply:
Two points about this:

1. Can there be value in enabling unsafe log for debug purposes even on newer clusters?
2. Why do we want to base that on the version when the cluster was created rather than the current version?